Changes

```diff
@@ -540,6 +540,9 @@ Working version
 - #12684: fix locations filename in AST produced by the `-pp` option
   (Gabriel Scherer, review by Florian Angeletti)
 
+- ???: disable common subexpression elimination for atomic loads
+  (Gabriel Scherer, review by ???, report by Vesa Karvonen)
```

Comment:
It was Carine Morel (@lyrm) who wrote [the test that triggered the bug](https://github.com/ocaml-multicore/saturn/pull/83#discussion_r1366768484).  Initially we assumed it was my mistake, but later I could not understand why the bug would be triggered (`Atomic.get` is supposed to have an acquire fence and that should prevent the reordering from happening), so I examined the compiler output and realized that the compiler had eliminated the `Atomic.get` instruction completely.